### PR TITLE
CFD-432 Implement OG permissions to the QP.

### PR DESCRIPTION
**Pull request by HelenaEksler**
https://issuetracker.amplexor.com/jira/browse/CFD-432

### CODE REVIEW

**Comment by HelenaEksler** [comment]
@nader77 Check, please.

File: capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/C4mRestfulEntityBaseNode.class.php [hunk 1]
```diff
@@ -9,6 +9,31 @@
...
+   *
+   * Set permissions to create group content based on user's OG permissions and
+   * group status value.
+   */
+  protected function checkEntityAccess($op, $entity_type, $entity) {
+    $account = $this->getAccount();
+    $resource_name = $this->getResourceName();
+    $group_id = $this->request['group'];
+    $group = node_load($group_id);
+    $wrapper = entity_metadata_wrapper('node', $group);
```
Comment (nader77) [inline capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/C4mRestfulEntityBaseNode.class.php]: Send ``$group_id`` directly to the wrapper, No need to ``node_load``.

**Comment by zero2one** [comment]
Non group members still get the quick post form, but they can't do anything with it.

* Log in with user "turing"
* Go to the Nobel Prize group "/nobelprize"
* QP is available
* Save results in error

=> The QP should not be shown.

Could be that this will be fixed when CFD-510 is ready & merged…

cfd432

**Comment by HelenaEksler** [comment]
@zero2one 
Hiding QP form based on the user's permissions is done in #709. This PR is only for implementing permissions. Because we are changing QP loading to "lazy loading" - don't want to get conflicts later. 
See [#709#files#diff-8f3e84672c8f835559f9eb60f7e32eecR79](https://github.com/capacity4dev/capacity4more/pull/709/files#diff-8f3e84672c8f835559f9eb60f7e32eecR79)

**Comment by HelenaEksler** [comment]
@MPParsley If you have time - review this also, please.

File: capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/C4mRestfulEntityBaseNode.class.php [hunk 2]
```diff
@@ -9,6 +9,30 @@
...
+    $group_id = $this->request['group'];
+    $wrapper = entity_metadata_wrapper('node', $group_id);
+    $group_status = $wrapper->c4m_og_status->value();
+    // Only platform admins can create group content if group status
+    // is NOT within the allowed groups array.
+    $allowed_groups = array(
+      'draft',
+      'published',
+    );
+    if (!in_array($group_status, $allowed_groups) && !user_access('administer site configuration', $account)) {
```
Comment (zero2one) [inline capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/C4mRestfulEntityBaseNode.class.php]: is this extra check still needed when https://issuetracker.amplexor.com/jira/browse/CFD-510 is implemented?

Comment (HelenaEksler) [inline capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/C4mRestfulEntityBaseNode.class.php]: I would prefer to leave it.

**Pull request merged**